backend/src/api/model/realm/mod.rs

```diff
@@ -268,4 +268,38 @@ impl Realm {
         // TODO: at some point, we want ACLs per realm
         context.user.is_moderator(&context.config.auth)
     }
+
+    /// Returns `true` if this realm somehow references the given node via
+    /// blocks. Currently, the following rules are used:
+    ///
+    /// - If `id` refers to a series: returns `true` if the realm has a series
+    ///   block with that series.
+    /// - If `id` refers to an event: returns `true` if the realm has a video
+    ///   block with that video OR if the realm has a series block with that
+    ///   events series.
+    /// - Otherwise, `false` is returned.
+    async fn references(&self, id: Id, context: &Context) -> ApiResult<bool> {
+        if let Some(event_key) = id.key_for(Id::EVENT_KIND) {
+            let query = "select count(*) \
```

Comment:
Using `exists` here might be more declarative and maybe even faster? Not sure. :thinking: